api/src/v2/Registration/Uploader/OrgDeputyshipUploader.php

```diff
@@ -198,12 +198,24 @@ private function handleClient(OrgDeputyshipDto $dto)
 //                $this->updated['clients'][] = $this->client->getId();
 //            }
 
-            // Temp fix for clients who have new named deputy in same organisation
-            if (!$this->clientHasSwitchedOrganisation($this->client)) {
-                if ($this->clientHasNewNamedDeputy($this->client, $this->namedDeputy)) {
-                    $this->client->setNamedDeputy($this->namedDeputy);
+            if (!$this->clientHasNewCourtOrder($this->client, $dto)) {
+                // Temp fix for deputies that have switched organisation and taken the client with them
+                if ($this->clientHasSwitchedOrganisation($this->client)) {
+                    if (!$this->clientHasNewNamedDeputy($this->client, $this->namedDeputy)) {
+                        $this->currentOrganisation->addClient($this->client);
+                        $this->client->setOrganisation($this->currentOrganisation);
+
+                        $this->updated['clients'][] = $this->client->getId();
+                    }
+                }
+
+                // Temp fix for clients who have new named deputy in same organisation
```

Comment:
If it's temporary, worth adding a TODO comment and referring to the ticket that will have the final fix. Most IDEs index TODOs in a helpful way.